File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int sz1 = word1.size() , sz2 = word2.size();
        int mnsz = min(sz1,sz2);
        string ans;
        for(int i=0;i<mnsz;i++)
        {
            ans+=word1[i];
            ans+=word2[i];
        }
        for(int i = mnsz; i<sz1;i++)
            ans+=word1[i];
        for(int i = mnsz; i<sz2;i++)
            ans+=word2[i];
        return ans;
    }
};

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder ans = new StringBuilder("");
        int n = Math.min(word1.length(),word2.length());
        for(int i = 0; i < n; i++) 
            ans.append(word1.charAt(i) + "" + word2.charAt(i)); 

        if(word1.length()<word2.length())
        {
            for(int i=n;i<word2.length();i++)ans.append(word2.charAt(i));
        }
        else
        {
            for(int i=n;i<word1.length();i++)ans.append(word1.charAt(i));
        }
        return ans.toString();
    }
}